### PR TITLE
adding a bunch of indexes to speed up queries

### PR DESCRIPTION
**Pull request by rabble**
I went through and found a bunch of queries that were not using indexes and then i created indexes for those queries.

> CREATE INDEX channel_assignments_idx_0039db ON channel_assignments(msg_ref);
> CREATE INDEX channel_assignments_idx_e349b0cd ON channel_assignments(chan_ref, msg_ref);
> CREATE INDEX reports_idx_14b8e58e ON reports(author_id, created_at DESC);
> CREATE INDEX pubs_index ON pubs(msg_ref);
> CREATE INDEX tangles_roots_and_msg_refs ON tangles(root, msg_ref);
> CREATE INDEX posts_root_mesgrefs ON posts(is_root, msg_ref);
> CREATE INDEX mention_feed_author_refs on mention_feed (feed_id, msg_ref);
> CREATE INDEX contacts_msg_ref ON contacts(msg_ref);
> CREATE INDEX contacts_state_and_author ON contacts(state, author_id);
> CREATE INDEX channel_assignments_msg_refs ON channel_assignments(msg_ref);
> CREATE INDEX channel_assignments_chan_ref_and_msg_ref ON channel_assignments(chan_ref, msg_ref);
> CREATE INDEX messages_idx_type_claimed_at ON messages(type, claimed_at);
> CREATE INDEX messages_idx_author_received ON messages(author_id, received_at DESC);
> CREATE INDEX messages_idx_author_type_sequence ON messages(author_id, type, sequence DESC);
> CREATE INDEX messages_idx_is_decrypted_hidden_claimed_at ON messages(is_decrypted, hidden, claimed_at);
> CREATE INDEX messages_idx_type_is_decrypted_hidden_author ON messages(type, is_decrypted, hidden, author_id);
> CREATE INDEX messages_idx_type_is_decrypted_hidden_claimed_at ON messages(type, is_decrypted, hidden, claimed_at);
> CREATE INDEX messages_idx_is_decrypted_hidden_author_claimed_at ON messages(is_decrypted, hidden, author_id, claimed_at);
> CREATE INDEX reports_author_created_at ON reports(author_id, created_at DESC);
> CREATE INDEX reports_msg_ref ON reports(msg_ref);
> CREATE INDEX reports_msg_ref_author ON reports(msg_ref, author_id;`

### CODE REVIEW

**Comment by mplorentz** [comment]
Looks like the unit tests are failing hard. I didn't get a chance to look into why today, I'll try again next week.

File: Source/GoBot/ViewDatabaseSchema.sql [hunk 1]
```diff
@@ -334,10 +355,16 @@ CREATE TABLE reports (
     FOREIGN KEY ( author_id ) REFERENCES authors( "id" )
 );
 
+CREATE INDEX reports_author_created_at ON reports(author_id, created_at DESC);
+CREATE INDEX reports_msg_ref ON reports(msg_ref);
+CREATE INDEX reports_msg_ref_author ON reports(msg_ref, author_id;
```
Comment (martindsq) [inline Source/GoBot/ViewDatabaseSchema.sql:360]: A closing parenthesis is missing here

File: Source/GoBot/ViewDatabase.swift [hunk 2]
```diff
@@ -263,6 +263,33 @@ class ViewDatabase {
...
+                        CREATE INDEX messages_idx_type_claimed_at ON messages(type, claimed_at);
+                        CREATE INDEX messages_idx_author_received ON messages(author_id, received_at DESC);
+                        CREATE INDEX messages_idx_author_type_sequence ON messages(author_id, type, sequence DESC);
+                        CREATE INDEX messages_idx_is_decrypted_hidden_claimed_at ON messages(is_decrypted, hidden, claimed_at);
+                        CREATE INDEX messages_idx_type_is_decrypted_hidden_author ON messages(type, is_decrypted, hidden, author_id);
+                        CREATE INDEX messages_idx_type_is_decrypted_hidden_claimed_at ON messages(type, is_decrypted, hidden, claimed_at);
+                        CREATE INDEX messages_idx_is_decrypted_hidden_author_claimed_at ON messages(is_decrypted, hidden, author_id, claimed_at);
+                        CREATE INDEX reports_author_created_at ON reports(author_id, created_at DESC);
+                        CREATE INDEX reports_msg_ref ON reports(msg_ref);
+                        CREATE INDEX reports_msg_ref_author ON reports(msg_ref, author_id;
```
Comment (martindsq) [inline Source/GoBot/ViewDatabase.swift:288]: A closing parenthesis is missing here

**Review by martindsq — CHANGES_REQUESTED** [review]
The issue with the unit tests is a missing closing parenthesis in the schema, so the database cannot be created.
Furthermore, I think the schema should not be updated because the chain of ifs that handle "database migrations" (db.userVersion) are executed after the initial database creation. There was upon a time when they weren't and we had to modify the schema, but it is not the case anymore.

**Comment by mplorentz** [comment]
I went ahead and fixed the syntax error and reverted the changes to `ViewDatabaseSchema.sql`.

I ran our performance tests and these indexes sped up some areas quite a bit. For 3% slower write speed we got a 45% speedup on the Discover tab query, and 96% speedup on the default Home feed algorithm!

**Review by martindsq — APPROVED** [review]

**Pull request merged**